modules/affile/tests/test_collection_comparator.c

```diff
@@ -53,78 +57,136 @@ void test_data_free(TestData *self)
-struct TestFileList delete_all = {{"test1", "test2"}, {NULL}, {"test1", "test2"}, {NULL}};
-struct TestFileList create_all = {{NULL}, {"test1", "test2"}, {NULL}, {"test1", "test2"}};
-struct TestFileList mixed = {{"test1", "test2", "test3"}, {"test1", "test4"}, {"test2", "test3"}, {"test4"}};
-
-ParameterizedTestParameters(params, multiple)
+ParameterizedTestParameters(collection_comparator, test_collection_changes)
 {
-  static struct TestFileList params[6];
-  params[0] = nothing_changed;
-  params[1] = last_file_deleted;
-  params[2] = first_file_deleted;
-  params[3] = delete_all;
-  params[4] = create_all;
-  params[5] = mixed;
-
-  return cr_make_param_array(struct TestFileList, params, sizeof (params) / sizeof (struct TestFileList));
+  static const struct TestFileList params[] =
+  {
+    {
+      0, "nothing_changed",
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{0, 0}, NULL, 0}},
+      {{{0, 0}, NULL, 0}}
+    },
+    {
+      1, "first_file_deleted",
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{2, 2}, "test2", 0}},
+      {{{1, 1}, "test1", 0}},
+      {{{0, 0}, NULL, 0}}
+    },
+    {
+      2, "last_file_deleted",
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{1, 1}, "test1", 0}},
+      {{{2, 2}, "test2", 0}},
+      {{{0, 0}, NULL, 0}}
+    },
+    {
+      3, "deleted_all",
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{0, 0}, NULL, 0}},
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{0, 0}, NULL, 0}}
+    },
+    {
+      4, "created_all",
+      {{{0, 0}, NULL, 0}},
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+      {{{0, 0}, NULL, 0}},
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}},
+    },
+    {
+      5, "mixed",
+      {{{1, 1}, "test1", 0}, {{2, 2}, "test2", 0}, {{3, 3}, "test3", 0}},
+      {{{1, 1}, "test1", 0}, {{4, 4}, "test4", 0}},
+      {{{2, 2}, "test2", 0}, {{3, 3}, "test3", 0}},
+      {{{4, 4}, "test4", 0}}
+    }
+  };
+
+  size_t nb_params = sizeof(params) / sizeof(const struct TestFileList);
+  return cr_make_param_array(const struct TestFileList, params, nb_params);
 }
 
-ParameterizedTest(struct TestFileList *tup, params, multiple)
+ParameterizedTest(const struct TestFileList *tup, collection_comparator, test_collection_changes)
 {
   TestData *data = test_data_new();
   CollectionComparator *comporator = collection_comparator_new();
   collection_comporator_set_callbacks(comporator,
                                       _handle_new,
                                       _handle_delete,
                                       data);
+  gint ndx = tup->ndx;
+  const CollectionComparatorEntry *curr_entry = NULL;
   gint i;
-  for (i = 0; tup->initial_files[i] != NULL; i++)
-    {
-      collection_comparator_add_initial_value(comporator, tup->initial_files[i]);
-    }
 
-  collection_comparator_start(comporator);
+  cr_log_info("=== %d. %s", ndx, tup->name);
```

Comment:
is this log fina, or was for testing, if the latter can you please give a more detailed description on what was logged?

Reply:
this simply logs the actual test data group index and name, basically just used for debugging, can be removed if you wish

Reply:
Ohh sorry my bad I thought that it was some left over, now I see it is the 
`"nothing_changed"`, `"last_file_deleted"`.